dcos/http.py

```diff
@@ -120,6 +137,15 @@ def _request(method,
     :rtype: Response
     """
 
+    if timeout is True:
+        if toml_config is None:
+            toml_config = config.get_config()
+
+        timeout = config.get_config_val("core.timeout", toml_config)
+        timeout = (DEFAULT_CONNECT_TIMEOUT, timeout or DEFAULT_READ_TIMEOUT)
```

Comment:
This sets the read timeout to `DEFAULT_READ_TIMEOUT` if `timeout` is set to 0. Is this the desired behaviour? Please add a test for this case.

Reply:
`timeout` to 0 doesn't really makes sense, it is not possible to set it like this through the config : 
```
$ dcos config set core.timeout 0                                                                      
Error: 0 is less than the minimum of 1
Path: timeout
Value: 0
```

Reply:
Great, that satisfies me.